frappe/app.py

```diff
@@ -397,21 +394,21 @@ def handle_exception(e):
 	return response
 
 
-def sync_database(rollback: bool) -> bool:
+def sync_database():
+	db = getattr(frappe.local, "db", None)
+	if not db:
+		# db isn't initialized, can't commit or rollback
+		return
```

Comment:
neither `after_rollback` or `after_commit` will run in this case. code in `after_request` / `after_response` will need to ensure a DB connection is available if the code needs it.